src/__internal__/focus-trap/focus-trap.spec.tsx

```diff
@@ -0,0 +1,779 @@
+import React, { useEffect, useRef, useState } from "react";
+import { render, screen, fireEvent } from "@testing-library/react";
+import { act } from "react-dom/test-utils";
+
+import FocusTrap, { FocusTrapProps } from "./focus-trap.component";
+import { RadioButton, RadioButtonGroup } from "../../components/radio-button";
+import { ModalContext } from "../../components/modal/modal.component";
+import { Select, Option } from "../../components/select";
+
+jest.useFakeTimers();
+
+interface MockComponentProps extends FocusTrapProps {
+  isAnimationComplete?: boolean;
+  triggerRefocusFlag?: boolean;
+  tabIndex?: number;
+  children: React.ReactNode;
+  shouldFocusFirstElement?: boolean;
+}
+
+const WRAPPER_ID = "test wrapper";
+const FIRST_ELEMENT = "first element";
+const BUTTON_ONE = "Test button One";
+const BUTTON_TWO = "Test button Two";
+const RADIO_LABEL_ONE = "Radio one";
+const RADIO_LABEL_TWO = "Radio two";
+const RADIO_GROUP_ONE = "Radio group one";
+const RADIO_GROUP_TWO = "Radio group two";
+
+const tabPress = () =>
+  fireEvent.keyDown(screen.getByTestId(WRAPPER_ID), {
+    key: "Tab",
+  });
+
+const shiftTabPress = () =>
+  fireEvent.keyDown(screen.getByTestId(WRAPPER_ID), {
+    key: "Tab",
+    shiftKey: true,
+  });
+
+const focusElement = (target: HTMLElement) => {
+  target.focus();
+  expect(target).toBeFocused();
+};
+
+const MockComponent = ({
+  children,
+  triggerRefocusFlag,
+  isAnimationComplete,
+  tabIndex,
+  shouldFocusFirstElement,
+  ...rest
+}: MockComponentProps) => {
+  const ref = useRef<HTMLDivElement>(null);
+  const [isDisabled, setIsDisabled] = useState(false);
+  const firstRef = useRef<HTMLButtonElement>(null);
+
+  return (
+    <ModalContext.Provider value={{ isAnimationComplete, triggerRefocusFlag }}>
+      <FocusTrap
+        wrapperRef={ref}
+        {...rest}
+        isOpen
+        focusFirstElement={shouldFocusFirstElement ? firstRef : undefined}
+      >
+        <div ref={ref} data-testid={WRAPPER_ID} tabIndex={tabIndex}>
+          {React.Children.map(children, (child) => {
+            const focusableChild = child as React.ReactElement;
+
+            if (focusableChild?.props?.id === "disable-on-focus") {
+              return React.cloneElement(focusableChild, {
+                onFocus: () => setIsDisabled(true),
+                disabled: isDisabled,
+              });
+            }
+
+            return child;
+          })}
+          {shouldFocusFirstElement && (
+            <button type="button" ref={firstRef}>
+              {FIRST_ELEMENT}
+            </button>
+          )}
+        </div>
+      </FocusTrap>
+    </ModalContext.Provider>
+  );
+};
+
+describe("FocusTrap", () => {
+  describe("triggerRefocusFlag", () => {
+    it("refocuses the last element that had focus within the trap when flag is set", () => {
+      const { rerender } = render(
+        <MockComponent autoFocus={false} triggerRefocusFlag={false}>
```

Comment:
Could we separate out the UI code here as another mock component, such as `<TriggerRefocusExample/>`, similarly to how the QAs have done with the refactored cypress tests? That way we could avoid duplicating the larger UI code examples.

Reply:
I'll see if I can just move everything into a single wrapper and avoid the need to have any but the one